src/app/handlers/new_member.handler.ts

```diff
@@ -1,32 +1,33 @@
-import { GuildMember, Role } from 'discord.js';
+import { GuildMember, Role, Collection } from 'discord.js';
 import { IContainer, IHandler } from '../../common/types';
 import { Maybe } from '../../common/types';
 
 export class NewMemberHandler implements IHandler {
-  private _unverifiedRole: Maybe<Role> = undefined;
-  private _unacknowledgedRole: Maybe<Role> = undefined;
-
+  private _roleCache: Record<string, Maybe<Role>> = {
+    Unacknowledged: undefined,
+    'Un verified': undefined,
+  };
   constructor(public container: IContainer) {}
 
   public async execute(member: GuildMember): Promise<void> {
-    if (!this._unacknowledgedRole) {
-      this._unacknowledgedRole = member.guild.roles
-        .filter((role) => role.name === 'Unacknowledged')
-        .first();
+    if (!this._roleCache.unacknowledged || !this._roleCache['Un verified']) {
+      Object.keys(this._roleCache).map((key: string) => {
+        this._roleCache[key] = member.guild.roles.filter((r) => r.name === key).first();
+      });
     }
-    member.addRole(this._unacknowledgedRole);
+
+    //Required to remove optional | undefined
+    if (!this._roleCache.Unacknowledged || !this._roleCache['Un verified']) {
```

Comment:
shoudlnt check unverified here

Reply:
the lack of Unverified role in teh guild shouldnt prevent us from adding unack role if it exists

Reply:
bump?

Reply:
My bad, resolved wrong convo and forgot ab it